src/librustc_driver/lib.rs

```diff
@@ -106,9 +106,32 @@ mod test;
 pub mod profile;
 pub mod driver;
 pub mod pretty;
-pub mod target_features;
 mod derive_registrar;
 
+pub mod target_features {
+    use syntax::ast;
+    use rustc::session::Session;
+    use syntax::symbol::Symbol;
+
+    /// Add `target_feature = "..."` cfgs for a variety of platform
+    /// specific features (SSE, NEON etc.).
+    ///
+    /// This is performed by checking whether a whitelisted set of
+    /// features is available on the target machine, by querying LLVM.
+    pub fn add_configuration(cfg: &mut ast::CrateConfig, sess: &Session) {
+        let tf = Symbol::intern("target_feature");
+
+        #[cfg(feature="llvm")]
+        for feat in ::rustc_trans::target_features(sess) {
```

Comment:
Can't you make the trait object return this? This is essential for pluggable LLVM.